gcs-fetcher/pkg/fetcher/fetcher.go

```diff
@@ -298,6 +317,17 @@ func (gf *Fetcher) fetchObjectOnce(ctx context.Context, j job, dest string, brea
 
 	r, err := gf.GCS.NewReader(ctx, j.bucket, j.object)
 	if err != nil {
+		// Check for AccessDenied failure here and return a useful error message on Stderr and exit immediately.
+		if strings.Contains(err.Error(), "<Code>AccessDenied</Code>") {
```

Comment:
I really dislike matching against the error message like this, but it seems like the structured error has already been unwrapped into a simple `error`, so the formal 403 code is gone. At least that's how it appeared to me.